pkg/test/framework/components/istio/operator.go

```diff
@@ -693,3 +708,132 @@ func deployCACerts(workDir string, env *kube.Environment, cfg Config) error {
 	}
 	return nil
 }
+
+func createIstiodEndpoint(discoveryAddress string, cfg Config, cluster resource.Cluster) error {
+	scopes.Framework.Infof("creating endpoints and service in cluster %s", cluster.Name())
+	svc := &kubeApiCore.Service{
+		ObjectMeta: kubeApiMeta.ObjectMeta{
+			Name:      istiodSvcName,
+			Namespace: cfg.SystemNamespace,
+		},
+		Spec: kubeApiCore.ServiceSpec{
+			Ports: []kubeApiCore.ServicePort{
+				{
+					Port:     15012,
+					Name:     "tcp-istiod",
+					Protocol: kubeApiCore.ProtocolTCP,
+				},
+				{
+					Name:     "tcp-webhook",
+					Protocol: kubeApiCore.ProtocolTCP,
+					Port:     15017,
+				},
+			},
+			ClusterIP: "None",
+		},
+	}
+	_, err := cluster.CoreV1().Services(cfg.SystemNamespace).Create(context.TODO(), svc, kubeApiMeta.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	eps := &kubeApiCore.Endpoints{
+		ObjectMeta: kubeApiMeta.ObjectMeta{
+			Name:      istiodSvcName,
+			Namespace: cfg.SystemNamespace,
+		},
+		Subsets: []kubeApiCore.EndpointSubset{
+			{
+				Addresses: []kubeApiCore.EndpointAddress{
+					{
+						IP: discoveryAddress,
+					},
+				},
+				Ports: []kubeApiCore.EndpointPort{
+					{
+						Name:     "tcp-istiod",
+						Protocol: kubeApiCore.ProtocolTCP,
+						Port:     15012,
+					},
+					{
+						Name:     "tcp-webhook",
+						Protocol: kubeApiCore.ProtocolTCP,
+						Port:     15017,
+					},
+				},
+			},
+		},
+	}
+
+	_, err = cluster.CoreV1().Endpoints(cfg.SystemNamespace).Create(context.TODO(), eps, kubeApiMeta.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	err = retry.UntilSuccess(func() error {
+		_, err := cluster.CoreV1().Services(cfg.SystemNamespace).Get(context.TODO(), istiodSvcName, kubeApiMeta.GetOptions{})
+		if err != nil {
+			return err
+		}
+		_, err = cluster.CoreV1().Endpoints(cfg.SystemNamespace).Get(context.TODO(), istiodSvcName, kubeApiMeta.GetOptions{})
+		if err != nil {
+			return err
+		}
+		return nil
+	}, retry.Timeout(20*time.Second), retry.Delay(2*time.Second))
+	return err
+}
+
+func deployWithRemoteKubeConfig(iopFile, remoteIopFile string, env *kube.Environment, cfg Config, i *operatorComponent) error {
+	var err error
+	// 1. install istiod Remote + base in remote cluster
```

Comment:
It seems like this is out of place. We should be installing the primary clusters first, and then the remotes. Can we do this when we do step 4?

Reply:
no, in this case, we need install remote cluster first as primary cluster is using KUBECONFIG from remote cluster.